api/server.py

```diff
@@ -1044,6 +1044,41 @@ def top_usernames_used():
         return flask_null_array_response()
 
 
+@app.route("/api/events/most-usernames-used-search", methods=["GET"])
+def top_usernames_used_search_query():
+    """
+    Get top usernames used according to module and search query
+
+    Returns:
+        JSON/Dict of top usernames used
+    """
+    module_name = get_value_from_request("module_name")
+    search_val = get_value_from_request("search_val")
+    module_query = [
+        group_by_ip_dest_and_username,
+        {
+            "$skip": fix_skip(
+                get_value_from_request("skip")
+            )
+        },
+        {
+            "$limit": fix_limit(
+                get_value_from_request("limit")
+            )
+        }
+    ]
+    if module_name:
+        module_query.insert(0, {"$match": {"$or": [ {'username': search_val, 'module_name': module_name}, {'ip_dest': search_val, 'module_name': module_name} ] } })
+    else:
+        module_query.insert(0, {"$match": {"$or": [ {'username': search_val}, {'ip_dest': search_val} ] } }
```

Comment:
bracket missing problem.
please test the code first and then send the PR.

Reply:
It must have been done unintentionally in the second commit( for removing whites paces) that I later squashed to maintain the branch. Point noted and will be taken care.